File: for-local-test/125.valid-palindrome.java

class Solution {
    public boolean isPalindrome(String s) {
        if(s == null || s.length() == 0){
            return true;
        }
        String str = s.toLowerCase().replaceAll("[^a-z0-9]", "");
        int lastIndex = str.length() - 1;
        int firstIndex = 0;
        while (firstIndex < lastIndex){
            if(str.charAt(firstIndex) != str.charAt(lastIndex)) return false;
            firstIndex++;
            lastIndex--;
        }
        return true;
    }
    
    public static void main(String[] args){
        Solution solution = new Solution();
        System.out.println(solution.isPalindrome("A man, a plan, a canal: Panama"));
        System.out.println(solution.isPalindrome("race a car"));
    }
}

File: for-local-test/python/125.valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        if s is None or len(s) == 0:
            return True
        cleaned = ''.join(ch.lower() for ch in s if ch.isalnum())
        first = 0
        last = len(cleaned) - 1
        while first < last:
            if cleaned[first] != cleaned[last]:
                return False
            first += 1
            last -= 1
        return True
    
if __name__ == "__main__":
    solution = Solution()
    print(solution.isPalindrome("A man, a plan, a canal: Panama"))  # Output: True
    print(solution.isPalindrome("race a car"))  # Output: False

        
            
